### PR TITLE
Add multiple inputs for lcm and gcd

### LINKED ISSUE #704: gcd with more than two arguments
Julia seems to support multiple arguments to gcd, We only support two, currently.

### PR DESCRIPTION
**Pull request by albinahlback**
Closes #704.

### CODE REVIEW

File: src/flint/fmpz.jl [hunk 1]
```diff
@@ -942,16 +942,23 @@ end
...
+function gcd(x::fmpz, y::fmpz, z::fmpz...)
+   d = fmpz()
    ccall((:fmpz_gcd, libflint), Nothing,
-         (Ref{fmpz}, Ref{fmpz}, Ref{fmpz}), z, x, y)
-   return z
+         (Ref{fmpz}, Ref{fmpz}, Ref{fmpz}), d, x, y)
+   length(z) == 0 && return d
+
+   for ix in 1:length(z)
+     tmp = d
```
Comment (wbhart) [inline src/flint/fmpz.jl:957]: Is this line necessary? Can't you just take the gcd of d and z[ix] below?

File: src/flint/fmpz.jl [hunk 2]
```diff
@@ -984,16 +991,23 @@ function gcd(x::Array{fmpz, 1})
...
+function lcm(x::fmpz, y::fmpz, z::fmpz...)
+   m = fmpz()
    ccall((:fmpz_lcm, libflint), Nothing,
-         (Ref{fmpz}, Ref{fmpz}, Ref{fmpz}), z, x, y)
-   return z
+         (Ref{fmpz}, Ref{fmpz}, Ref{fmpz}), m, x, y)
+   length(z) == 0 && return m
+
+   for ix in 1:length(z)
+     tmp = m
```
Comment (wbhart) [inline src/flint/fmpz.jl:1006]: Same here.

File: src/flint/fmpz.jl (hunk 1 above)
Comment (albinahlback) [reply src/flint/fmpz.jl]: Sorry, of course. Thank you!

**Comment by wbhart** [comment]
I'll merge this when green. Thanks again for the contribution.

**Pull request merged**